heartbeat: get gain from majestic metrics if possible

viktorxda opened the pull request.

themactep commented:
Can you make majestic to return only one value if requested with a parameter? Otherwise we have to use resources to generate, store and parse a lot of unnecessary information.
```
https://127.0.0.1/metrics/isp-value
567
```

viktorxda commented:
That should be possible, the current idea is to query the needed parameter with 'value':
```
http://192.168.1.20/metrics?value=isp_again
131072
```

themactep commented:
That'll do. But let's move majestic querying higher and do the alternative methods after that, only if the value is not there (in case there is no majestic or its web is disabled hence no metrics).

viktorxda commented:
Changed the following:
- moved the majestic check to the start
- query the regular check only if `daynight_value` is empty
- added a check for sigmastar, the sar driver is not available for all devices

themactep commented on files/var/www/cgi-bin/j/heartbeat.cgi:
```diff
@@ -10,18 +10,26 @@ else
+			daynight_value=$(imp-control.sh gettotalgain)
+			;;
+		sigmastar)
+			if [ -e /sys/devices/virtual/mstar/sar/channel ]; then
```
this `if` requres `else` with `daynight_value=-1`, otherwise sigmastar devices with no sar file get no value assigned.

viktorxda replied:
This has been addressed with the latest change.
The new changes for the metrics should be available now for testing.

themactep commented:
Great! Then let's merge it. Thank you.
